portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,25 +51,33 @@ public DataServlet() {
 
   @Override
   protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-    String comment = req.getParameter("comment");
+    String comment = req.getParameter(COMMENT_KIND);
     if (comment != null && !comment.equals("")) {
       // do nothing if the comment field is empty
-      Entity commentEntity = new Entity("comment");
-      commentEntity.setProperty("value", comment);
+      Entity commentEntity = new Entity(COMMENT_KIND);
+      commentEntity.setProperty(COMMENT_VALUE_KEY, comment);
       datastore.put(commentEntity);
     }
-    resp.sendRedirect("/index.html");
+    resp.sendRedirect(HOME_PAGE_PATH);
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("comment");
-    PreparedQuery results = datastore.prepare(query);
-    ArrayList<String> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      comments.add((String)entity.getProperty("value"));
+    String maxCommentsString = request.getParameter(MAX_COMMENTS_PARAM);
+    Integer maxComments = null;
```

Comment:
maxComments can be an int instead of Integer? This is more natural, and can be achieved via:

int maxComments = DEFAULT_MAX_COMMENTS;
try {
  maxComments = Integer.parseInt(maxCommentsString);
} catch (NumberFormatException e) {
  // do nothing (maybe log an error in prod, who knows)
}